Zotero/Extensions/MD5+Url.swift

```diff
@@ -44,3 +44,23 @@ func md5(from url: URL) -> String? {
         return nil
     }
 }
+
+var cachedMD5AndModificationDateByURL: [URL: (String, Date)] = [:]
+func cachedMD5(from url: URL, using fileManager: FileManager) -> String? {
+    let newModificationDate: Date
+    if let attributes = try? fileManager.attributesOfItem(atPath: url.path), let modificationDate = attributes[.modificationDate] as? Date {
+        newModificationDate = modificationDate
+    } else {
+        newModificationDate = .distantPast
+    }
+    if let (cachedMd5, cachedModificationDate) = cachedMD5AndModificationDateByURL[url], newModificationDate <= cachedModificationDate {
```

Comment:
It shouldn't matter, but why `newModificationDate <= cachedModificationDate`? Why not equality only?

Reply:
Hmmm, you are right, to be on the safe side, any change should be enough for recomputing the hash. Check https://github.com/zotero/zotero-ios/pull/979/commits/82c0443562bee4ba13578f5e89f18b5cee244472

Maybe we should add a size comparison as well?

Reply:
I guess if the size changes the modification date should change as well, so it might be redundant. But it wouldn't hurt anyone.